projects/driver_controls/inc/stm32f0xx/ads1015.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+// Module for using ads1015
+// GPIO and Interrupt, I2C should be initiated
+// The user also needs to create a struct ADS1015Data which would persist accross functions
+// start the ads1015 using ads1015_init and then configure channels using ads1015_config_channel
+// read raw and converted values using ads1015_read_raw and ads1015_read_converted.
+#include <stdbool.h>
+#include "gpio.h"
+#include "i2c.h"
+#include "status.h"
+
+typedef enum {
+  ADS1015_ADDRESS_GND = 0,
+  ADS1015_ADDRESS_VDD,
+  ADS1015_ADDRESS_SDA,
+  ADS1015_ADDRESS_SCL,
+  NUM_ADS1015_ADDRESSES,
+} ADS1015Address;
+
+typedef enum {
+  ADS1015_CHANNEL_0 = 0,
+  ADS1015_CHANNEL_1,
+  ADS1015_CHANNEL_2,
+  ADS1015_CHANNEL_3,
+  NUM_ADS1015_CHANNELS,
+} ADS1015Channel;
+
+typedef void (*ADS1015Callback)(const GPIOAddress *address, void *context);
```

Comment:
Why do you return a GPIOAddress? What's the intended usage of this callback?

Reply:
Do you mean why I am setting it as a parameter? Ah! I just realized I am not registering this callback(user's) on GPIO pin, so it needs only the context. Correct?

Reply:
Yeah, consider it from the user's perspective - if you were using this module, what information would be useful in the callback? In this case, I don't think the GPIO pin is useful, but I think there's other information that might be.

For example, what if you used the same callback for all 4 channels?

Reply:
I'm guessing channel would be a good parameter?